src/app/wallets/add-invoice-for-wallet.ts

```diff
@@ -0,0 +1,113 @@
+import { toSats } from "@domain/bitcoin"
+import { invoiceExpirationForCurrency } from "@domain/bitcoin/lightning"
+import { WalletInvoiceFactory } from "@domain/wallet-invoices/wallet-invoice-factory"
+import { LndService } from "@services/lnd"
+import {
+  WalletInvoicesRepository,
+  UsersRepository,
+  AccountsRepository,
+} from "@services/mongoose"
+
+export const addInvoiceForSelf = async ({
+  walletId,
+  amount,
+  memo = "",
+}: AddInvoiceSelfArgs): Promise<LnInvoice | ApplicationError> => {
+  const walletInvoiceFactory = WalletInvoiceFactory(walletId)
+  return baseAddInvoiceForWallet({
+    amount,
+    memo,
+    walletInvoiceCreateFn: walletInvoiceFactory.create,
+  })
+}
+
+export const addInvoiceNoAmountForSelf = async ({
+  walletId,
+  memo = "",
+}: AddInvoiceNoAmountSelfArgs): Promise<LnInvoice | ApplicationError> => {
+  const walletInvoiceFactory = WalletInvoiceFactory(walletId)
+  return baseAddInvoiceForWallet({
+    amount: toSats(0),
+    memo,
+    walletInvoiceCreateFn: walletInvoiceFactory.create,
+  })
+}
+
+export const addInvoiceForRecipient = async ({
+  username,
```

Comment:
This arg should be `recipient` or `recipientUserName`

Reply:
Changing to `recipient` because that's already used in some places